architecture/device_tree.rst

```diff
@@ -23,34 +23,61 @@ shared across the boot stages.
     - Boot loader stages may load a device tree structure in memory for all boot
       stage to get platform configuration from. If such device tree data are to
       be accessed by the non-secure world, they shall be located in non-secure
-      memory.
+      memory. Secure world may used its content during the OP-TEE core
+      initialization.
 
     - Boot loader stages may load a device tree structure in secure memory for
       the benefit of the secure world only. Such device tree blob shall be
-      located in secure memory.
+      located in secure memory. Secure world could used its content but this
+      is currently not implemented in the OP-TEE latest release.
 
     - OP-TEE core can also embedded a device tree structure to describe the
       platform.
 
-    - Non-secure world can embed a device tree structure and/or rely on a device
-      tree structure loaded by the secure world, being an early boot stage
-      and/or OP-TEE core.
+    - Non-secure world can embed its own device tree structure(s) and/or
+      rely on a device tree structure loaded by the secure world during
+      its initialization which happen before non-secure world is booted.
 
 Obviously the non-secure world will not be able to access a device tree image
-located on a secure memory which non-secure world as no access to.
+located in a secure memory which non-secure world as no access to.
 
-Early boot device tree argument
+When OP-TEE core is built with `CFG_DT=y`, non-secure and secure device trees
```

Comment:
rst-syntax uses double back-quote to make "code sections" (single gives italic), so please add that around `CFG_DT=y`.